src/Type/Regex/RegexAstWalkResult.php

```diff
@@ -0,0 +1,104 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Regex;
+
+/** @immutable */
+final class RegexAstWalkResult
+{
+
+	/**
+	 * @param array<int, RegexCapturingGroup> $capturingGroups
+	 * @param list<string> $markVerbs
+	 */
+	public function __construct(
+		private int $alternationId,
+		private int $captureGroupId,
+		private array $capturingGroups,
+		private array $markVerbs,
+	)
+	{
+	}
+
+	public static function createEmpty(): self
+	{
+		return new self(
+			-1,
+			100,
+			[],
+			[],
+		);
+	}
+
+	public function nextAlternationId(): self
```

Comment:
name it better, it does not return `int`

Reply:
it returns a "copy" of the current object, which contains the next alternation id.

I think the name fits.